intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/base/problems/PluginFileError.kt

```diff
@@ -17,15 +16,44 @@ class IncorrectPluginFile(
   private val expectedFileType: String
 ) : PluginFileError() {
   override val message
-    get() = "Incorrect plugin file $fileName. Must be a $expectedFileType"
+    get() = "The plugin archive contains an unexpected file $fileName: it must be a $expectedFileType."
+}
+
+class IncorrectZipOrJarFile(
+  private val fileName: String
+) : PluginFileError() {
+  override val message
+    get() = "The plugin archive contains an unexpected file $fileName: it must be .zip, .jar or a directory."
 }
 
 class UnableToExtractZip : PluginFileError() {
   override val message
-    get() = "Unable to extract plugin zip file"
+    get() = "The plugin archive file cannot be extracted: please use Gradle or Plugin DevKit to build the plugin distribution."
```

Comment:
But maybe the message is also incorrect

Reply:
Well, since hint and url are not ready to by public, I would like to skip adding it for now. What's wrong with the message? It was approved by Natasha.

Reply:
Should plugin developers use Gradle or Plugin DevKit if they developer plugin for Space or Teamcity?

Reply:
Make sense. I'll ask Natatsha :)